tachyon/zk/plonk/vanishing/vanishing_constructed.h

```diff
@@ -0,0 +1,73 @@
+// Copyright 2020-2022 The Electric Coin Company
+// Copyright 2022 The Halo2 developers
+// Use of this source code is governed by a MIT/Apache-2.0 style license that
+// can be found in the LICENSE-MIT.halo2 and the LICENCE-APACHE.halo2
+// file.
+
+#ifndef TACHYON_ZK_PLONK_VANISHING_VANISHING_CONSTRUCTED_H_
+#define TACHYON_ZK_PLONK_VANISHING_VANISHING_CONSTRUCTED_H_
+
+#include <utility>
+#include <vector>
+
+#include "tachyon/zk/base/entities/entity_ty.h"
+#include "tachyon/zk/plonk/vanishing/vanishing_committed.h"
+
+namespace tachyon::zk {
+
+template <EntityTy EntityType, typename PCSTy>
+class VanishingConstructed;
+
+template <typename PCSTy>
+class VanishingConstructed<EntityTy::kProver, PCSTy> {
+ public:
+  using F = typename PCSTy::Field;
+  using Poly = typename PCSTy::Poly;
+  using ExtendedPoly = typename PCSTy::ExtendedPoly;
+
+  VanishingConstructed() = default;
+  VanishingConstructed(std::vector<Poly>&& h_pieces, Poly&& h_blinds_poly,
+                       VanishingCommitted<EntityTy::kProver, PCSTy>&& committed)
+      : h_pieces_(std::move(h_pieces)),
+        h_blinds_poly_(std::move(h_blinds_poly)),
+        committed_(std::move(committed)) {}
+
+  const std::vector<Poly>& h_pieces() const { return h_pieces_; }
+  const Poly& h_blinds_poly() const { return h_blinds_poly_; }
+
+  VanishingCommitted<EntityTy::kProver, PCSTy>&& TakeCommitted() && {
+    return std::move(committed_);
+  }
+
+ private:
+  std::vector<Poly> h_pieces_;
+  Poly h_blinds_poly_;
```

Comment:
Why was `h_blinds` created as `Poly` instead of `std::vector<F>`?

Reply:
To make the code more intuitive, I've refactored [this](https://github.com/kroma-network/halo2/blob/7d0a36990452c8e7ebd600de258420781a9b7917/halo2_proofs/src/plonk/vanishing/prover.rs#L138-L144) part of code as an evaluation of a `h_blinds_poly` on `x_n`.